src/components/mainpage/sponsors/sponsors.js

```diff
@@ -0,0 +1,117 @@
+import React from "react"
+import { graphql, useStaticQuery } from "gatsby"
+import SingleSponsor from "./single-sponsor"
+import sponsorsStyles from "../../../styles/sponsors.module.css"
+import Carousel from "../../utils/carousel"
+
+const getSponsors = (data, type) => {
+  return data.allMarkdownRemark.edges
+    .filter(({ node }) => node.frontmatter.type === type)
+    .map(({ node }) => (
+      <SingleSponsor
+        key={node.frontmatter.name}
+        sponsor={node.frontmatter}
+        type={type}
+      />
+    ))
+}
+
+const Sponsors = () => {
+  const data = useStaticQuery(graphql`
+    query SponsorsQuery {
+      allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/sponsors/" } }) {
+        edges {
+          node {
+            id
+            frontmatter {
+              name
+              type
+              img {
+                childImageSharp {
+                  fluid {
+                    ...GatsbyImageSharpFluid
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  return (
+    <section id="sponsors" className={sponsorsStyles.sponsorsSection}>
+      <h2 className={sponsorsStyles.h2}>
+        Sponsors
+        <hr className={sponsorsStyles.headingLine} />
+      </h2>
+
+      <div className={sponsorsStyles.mainSponsor}>
+        <div className={sponsorsStyles.sponsorsType}>Main</div>
+        <div className={sponsorsStyles.mainWrapper}>
+          {getSponsors(data, "main")}
+        </div>
+      </div>
+
+      <div
+        className={[
+          sponsorsStyles.otherSponsors,
+          sponsorsStyles.goldWrapper,
+        ].join(" ")}
+      >
+        <div
```

Comment:
Perhaps each of these divs could be extracted to a component? The data would be stored in an array and then we would map through it and display the sponsors.

Reply:
Created a component `Sponsor Section` to solve this.